File: steam_game_lookup.py
#!/usr/bin/env python3

import requests
import os

def load_steam_vars(filename="steam_vars.txt"):
    """Load Steam API key and Steam ID from a file."""
    steam_vars = {}
    if not os.path.exists(filename):
        raise FileNotFoundError(f"Configuration file '{filename}' not found.")

    with open(filename, "r") as file:
        for line in file:
            key, value = line.strip().split("=", 1)
            steam_vars[key] = value

    return steam_vars.get("STEAM_API_KEY"), steam_vars.get("STEAM_ID")

def get_owned_games(api_key, steam_id):
    """Retrieve the list of owned games for a user."""
    url = f"http://api.steampowered.com/IPlayerService/GetOwnedGames/v1/"
    params = {
        "key": api_key,
        "steamid": steam_id,
        "include_appinfo": True
    }
    response = requests.get(url, params=params)

    if response.status_code == 200:
        return response.json().get("response", {}).get("games", [])
    else:
        print("Error fetching game list:", response.text)
        return []

def find_game_id(game_name, games):
    """Find the game ID by game name."""
    for game in games:
        if game_name.lower() in game.get("name", "").lower():
            return game["appid"]
    return None

def find_game_name(game_id, games):
    """Find the game name by game ID."""
    for game in games:
        if game["appid"] == game_id:
            return game["name"]
    return None

def main():
    api_key, steam_id = load_steam_vars()
    if not api_key or not steam_id:
        print("Error: Steam API Key or Steam ID is missing in steam_vars.txt")
        return

    games = get_owned_games(api_key, steam_id)
    if not games:
        print("No games found or error fetching games.")
        return

    search = input("Enter a game name or game ID: ")

    if search.isdigit():
        search_id = int(search)
        result = find_game_name(search_id, games)
        print(f"Game Name: {result}" if result else "Game not found.")
    else:
        result = find_game_id(search, games)
        print(f"Game ID: {result}" if result else "Game not found.")

if __name__ == "__main__":
    main()

File: steam_game_lookup.go
package main

import (
    "bufio"
    "encoding/json"
    "fmt"
    "net/http"
    "os"
    "strconv"
    "strings"
)

// SteamVars holds the Steam API key and Steam ID
type SteamVars struct {
    APIKey  string
    SteamID string
}

// Game represents a Steam game
type Game struct {
    AppID int    `json:"appid"`
    Name  string `json:"name"`
}

// LoadSteamVars loads Steam API key and Steam ID from a file
func LoadSteamVars(filename string) (SteamVars, error) {
    file, err := os.Open(filename)
    if err != nil {
        return SteamVars{}, fmt.Errorf("configuration file '%s' not found", filename)
    }
    defer file.Close()

    vars := SteamVars{}
    scanner := bufio.NewScanner(file)
    for scanner.Scan() {
        line := scanner.Text()
        parts := strings.SplitN(line, "=", 2)
        if len(parts) == 2 {
            key, value := parts[0], parts[1]
            if key == "STEAM_API_KEY" {
                vars.APIKey = value
            } else if key == "STEAM_ID" {
                vars.SteamID = value
            }
        }
    }

    if err := scanner.Err(); err != nil {
        return SteamVars{}, err
    }

    return vars, nil
}

// GetOwnedGames retrieves the list of owned games for a user
func GetOwnedGames(apiKey, steamID string) ([]Game, error) {
    url := fmt.Sprintf("https://api.steampowered.com/IPlayerService/GetOwnedGames/v1/?key=%s&steamid=%s&include_appinfo=true", apiKey, steamID)
    resp, err := http.Get(url)
    if err != nil {
        return nil, fmt.Errorf("error fetching game list: %v", err)
    }
    defer resp.Body.Close()

    if resp.StatusCode != http.StatusOK {
        return nil, fmt.Errorf("error fetching game list: %s", resp.Status)
    }

    var result struct {
        Response struct {
            Games []Game `json:"games"`
        } `json:"response"`
    }
    if err := json.NewDecoder(resp.Body).Decode(&result); err != nil {
        return nil, fmt.Errorf("error decoding response: %v", err)
    }

    return result.Response.Games, nil
}

// FindGameID finds the game ID by game name
func FindGameID(gameName string, games []Game) int {
    for _, game := range games {
        if strings.Contains(strings.ToLower(game.Name), strings.ToLower(gameName)) {
            return game.AppID
        }
    }
    return 0
}

// FindGameName finds the game name by game ID
func FindGameName(gameID int, games []Game) string {
    for _, game := range games {
        if game.AppID == gameID {
            return game.Name
        }
    }
    return ""
}

func main() {
    // Load Steam API key and Steam ID
    steamVars, err := LoadSteamVars("steam_vars.txt")
    if err != nil {
        fmt.Println("Error:", err)
        return
    }

    if steamVars.APIKey == "" || steamVars.SteamID == "" {
        fmt.Println("Error: Steam API Key or Steam ID is missing in steam_vars.txt")
        return
    }

    // Fetch owned games
    games, err := GetOwnedGames(steamVars.APIKey, steamVars.SteamID)
    if err != nil {
        fmt.Println("Error:", err)
        return
    }

    if len(games) == 0 {
        fmt.Println("No games found or error fetching games.")
        return
    }

    // Handle user input
    var search string
    if len(os.Args) > 1 {
        search = strings.Join(os.Args[1:], " ")
    } else {
        fmt.Print("Enter a game name or game ID: ")
        fmt.Scanln(&search)
    }

    if gameID, err := strconv.Atoi(search); err == nil {
        // Search by game ID
        result := FindGameName(gameID, games)
        if result != "" {
            fmt.Printf("Game Name: %s\n", result)
        } else {
            fmt.Println("Game not found.")
        }
    } else {
        // Search by game name
        result := FindGameID(search, games)
        if result != 0 {
            fmt.Printf("Game ID: %d\n", result)
        } else {
            fmt.Println("Game not found.")
        }
    }
}